sources/api/pluto/src/api.rs

```diff
@@ -1,76 +1,151 @@
-    async fn get_settings() -> Result<model::Settings> {
-        let uri = constants::API_SETTINGS_URI;
-        let (_status, response_body) =
-            apiclient::raw_request(constants::API_SOCKET, uri, "GET", None)
-                .await
-                .context(ApiClientSnafu { uri })?;
+#[derive(Deserialize)]
+struct View {
+    pub aws: Option<AwsInfo>,
+    pub kubernetes: Option<Kubernetes>,
+}
 
-        serde_json::from_str(&response_body).context(SettingsJsonSnafu)
-    }
+#[derive(Deserialize)]
+struct SettingsView {
+    pub os: Os,
+    pub settings: View,
+}
 
-    /// Gets the info that we need to know about the EKS cluster from the Bottlerocket API.
-    pub(crate) async fn get_aws_k8s_info() -> Result<AwsK8sInfo> {
-        let settings = get_settings().await?;
-        Ok(AwsK8sInfo {
-            region: settings.aws.and_then(|a| a.region).map(|s| s.into()),
-            cluster_name: settings
-                .kubernetes
-                .as_ref()
-                .and_then(|k| k.cluster_name.clone())
-                .map(|s| s.into()),
-            cluster_dns_ip: settings.kubernetes.and_then(|k| k.cluster_dns_ip),
-        })
-    }
+#[derive(Debug, Snafu)]
+pub(crate) enum Error {
+    #[snafu(display("Failed to call apiclient: {}", source))]
+    CommandFailure { source: std::io::Error },
+    #[snafu(display("apiclient execution failed: {}", reason))]
+    ExecutionFailure { reason: String },
+    #[snafu(display("Deserialization of configuration file failed: {}", source))]
+    Deserialize {
+        #[snafu(source(from(serde_json::Error, Box::new)))]
+        source: Box<serde_json::Error>,
+    },
 }
 
-/// This dummy code is compiled when the `sources` workspace is being compiled for non `aws-k8s-*`
-/// variants.
-#[cfg(not(variant_family = "aws-k8s"))]
-mod inner {
-    use super::*;
-    use snafu::Snafu;
+pub(crate) async fn client_command<I, S>(args: I) -> Result<Vec<u8>>
+where
+    I: IntoIterator<Item = S>,
+    S: AsRef<OsStr>,
+{
+    let result = Command::new("/usr/bin/apiclient")
+        .args(args)
+        .output()
+        .await
+        .context(CommandFailureSnafu)?;
 
-    #[derive(Debug, Snafu)]
-    pub(crate) enum Error {
-        #[snafu(display(
-            "The get_aws_k8s_info function is only compatible with aws-k8s variants"
-        ))]
-        WrongVariant,
-    }
+    ensure!(
+        result.status.success(),
+        ExecutionFailureSnafu {
+            reason: String::from_utf8_lossy(&result.stderr)
+        }
+    );
+
+    Ok(result.stdout)
+}
 
-    pub(crate) async fn get_aws_k8s_info() -> Result<AwsK8sInfo> {
-        WrongVariantSnafu.fail()
-    }
+/// Gets the info that we need to know about the EKS cluster from the Bottlerocket API.
+pub(crate) async fn get_aws_k8s_info() -> Result<AwsK8sInfo> {
+    let view_str = client_command(&[
+        "get",
+        "os.variant-id",
+        "settings.aws.region",
```

Comment:
You would also need to retrieve the proxy server settings:
```
settings.network.https-proxy
settings.network.no-proxy
```

And ensure that they are set in the environment where they can be found by `proxy::fetch_proxy_env`.

This is something `sundog` does on behalf of settings generators.

Note that `sundog` has [k8s-specific no_proxy logic](https://github.com/bottlerocket-os/bottlerocket/blob/fcf71a47c0ff005327ecde870d8a70877fc89196/sources/api/sundog/src/main.rs#L281), but it appears this is not needed since we don't connect to those hosts.